Dockerfile.ci

```diff
@@ -0,0 +1,40 @@
+ARG BUILDER_IMAGE=registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19
+ARG BASE_IMAGE=registry.ci.openshift.org/ocp/builder:rhel-9-base-openshift-4.19
+
```

Comment:
Can we just use Dockerfile.Ocp?

it would be best if we could at least use the same image that influx builds. That way we don’t get weird issues with tests passing in Prow but failing in influx.

Reply:
We could also just switch Konflux to use Dockerfile.CI

Reply:
Konflux can't pull images from `registry.ci.openshift.org` and Prow can't pull images from `brew`.

Reply:
Apart from the L:1 and L:2, these images should be identical (maybe we can add a script to check their equality)

Reply:
Painful!

Reply:
yeah, I agree

Reply:
Konflux can pull images from registry.ci. I originally added a secret for that. But I think we want brew as that gives us the special golang build.

Reply:
> But I think we want brew as that gives us the special golang build.

That is correct. We can try to merge them in the future. But for now, having 2 different dockerfiles shouldn't create any issues to us.

Reply:
Could you post on test platform to see if Prow could get access to brew?

Reply:
Asked on Slack, thank you for review.

Reply:
They said Dockerfile.ocp may work on Prow. What about we merge this PR and I test using `Dockerfile.ocp` and if it works, I'll open a PR to remove `Dockerfile.ci`?. Does this sound ok?.

Reply:
I've tested Dockerfile.ocp on Prow in https://github.com/openshift/release/pull/61443 and it can't pull images from brew.registry.redhat.io